cetmix_tower_server/models/cx_tower_plan_line.py

```diff
@@ -47,6 +49,49 @@ class CxTowerPlanLine(models.Model):
         help="Conditions under which this Flight Plan Line "
         "will be launched. e.g.: {{ odoo_version}} == '14.0'",
     )
+    variable_ids = fields.Many2many(
+        comodel_name="cx.tower.variable",
+        relation="cx_tower_plan_line_variable_rel",
+        column1="plan_line_id",
+        column2="variable_id",
+        string="Variables",
+        compute="_compute_variable_ids",
+        store=True,
+    )
+
+    @api.depends("condition")
+    def _compute_variable_ids(self):
+        """
+        Compute and assign the variable_ids field by searching for variables
+        based on references from the condition field.
+        If no variables are found, the field is cleared.
+        """
+        for record in self:
+            all_references = set()
+            if record.condition:
+                all_references.update(record.get_variables_from_code(record.condition))
+
+            if all_references:
+                variables = self.env["cx.tower.variable"].search(
+                    [("reference", "in", list(all_references))]
+                )
+                record.variable_ids = [(6, 0, variables.ids)]
+            else:
+                record.variable_ids = [(5, 0, 0)]
+
+    def get_variables_from_code(self, code):
```

Comment:
Why not to use [get_variables_from_code()](https://github.com/cetmix/cetmix-tower/blob/4bf482c65c63fcd97417a08bfac6ad6ab4b8a317/cetmix_tower_server/models/cx_tower_template_mixin.py#L34) function from cx.tower.temlate.mixin?

```
self.env["cx.tower.template.mixin"].get_variables_from_code(your_code)
```